marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -176,12 +176,25 @@ function setArtifact(artifactType, artifactName, artifact) {
 }
 
 function validateArtifact(artifactType, artifactName, artifact) {
+  if(!validateArtifactName(artifactName)){
+    let message = `Invalid ${artifactType}`;
+    if(!(artifactType == "flow" || artifactType == "stepDefinition")){
+      message += " step";
+    }
+    message += ` name: '${artifactName}'; it must start with a letter and can contain letters, numbers, hyphens and underscores only.`
+    httpUtils.throwBadRequest(message);
```

Comment:
This is preventing the concrete artifact from doing its own validation. That can result in the user first getting this name validation error, and then fixing it, only to get another error next. Ideally, the user gets all errors at once so they only have to do one set of fixes. Is the UI able to handle many errors, or are we restricted for now to a single error message? That's fine if so, but likely something to improve in the near future.

Reply:
It looks like HC UI will do client side validation based on comments in https://project.marklogic.com/jira/browse/DHFPROD-6454

Reply:
Argh. The way to get rapid feedback isn't by duplicating knowledge across tiers, it's by immediately asking the backend to validate something after the user changes it.